rescript/ncbi.py

```diff
@@ -0,0 +1,141 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import requests
+from xmltodict import parse
+from pandas import DataFrame
+from q2_types.feature_data import DNAIterator
+from skbio import DNA
+from qiime2 import Metadata
+
+
+def get_ncbi_data(query: str = None, accession_ids: Metadata = None) -> (
+        DNAIterator, DataFrame):
+    if query is None and accession_ids is None:
+        raise ValueError('Query or accession_ids must be supplied')
+
+    if query:
+        seqs, taxids = get_nuc_for_query(query)
+
+    if accession_ids:
+        accs = accession_ids.get_ids()
+        if query and seqs:
```

Comment:
should we just allow a query or list of accession IDs instead of a mix of both?

Reply:
We've had this conversation offline, but for posterity, there is a use case here.

What if you wanted to download the RefSeq Targeted Loci for 16S (query: '33175[BioProject] OR 33317[BioProject]') supplemented with all the sequences they used for the Stirrups custom DB (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3535711/bin/1471-2164-13-S8-S17-S4.xls)?